lib/sitemap/generator.rb

```diff
@@ -0,0 +1,150 @@
+module Sitemap
+  class Generator
+    def initialize(search_config, uploader, timestamp)
+      @search_config    = search_config
+      @search_client    = Services.elasticsearch(hosts: search_config.base_uri, timeout: 10)
+      @uploader         = uploader
+      @timestamp        = timestamp
+    end
+
+    def run
+      create_sitemap_index(create_sitemaps(get_all_documents))
+    end
+
+    def create_sitemaps(enumerator)
+      enumerator.with_index.map do |documents, index|
+        batch_number = index + 1
+        documents.unshift(homepage) if batch_number == 1
```

Comment:
Instead of `unshift`ing the homepage, would it be less confusing for line 76 to be `chunk = [homepage]`?  The problem with doing it here is that the first sitemap file will have 1 more entry than all the others, which could be confusing (especially if we decide we want all the sitemap files to have 50,000 entries, as that's the maximum).

Reply:
I really like that, let's do it.